pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -18,60 +18,69 @@ package plugin
 
 import (
 	"context"
-	"errors"
 	"fmt"
-	"io"
-	"net"
 	"time"
 
 	"google.golang.org/grpc"
 	grpccodes "google.golang.org/grpc/codes"
-	"google.golang.org/grpc/credentials/insecure"
 	grpcstatus "google.golang.org/grpc/status"
 	"k8s.io/klog/v2"
 
 	drapbv1alpha2 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
-	drapb "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
+	drapbv1alpha3 "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
 )
 
 const PluginClientTimeout = 10 * time.Second
 
-// Strongly typed address.
-type draAddr string
+type versionedClient struct {
+	version string
+}
 
-// draPluginClient encapsulates all dra plugin methods.
-type draPluginClient struct {
-	pluginName        string
-	addr              draAddr
-	nodeClientCreator nodeClientCreator
+func (v *versionedClient) v1alpha2NodePrepareResource(ctx context.Context, conn *grpc.ClientConn, req *drapbv1alpha2.NodePrepareResourceRequest) (*drapbv1alpha2.NodePrepareResourceResponse, error) {
+	nodeClient := drapbv1alpha2.NewNodeClient(conn)
+	return nodeClient.NodePrepareResource(ctx, req)
 }
 
-var _ drapb.NodeClient = &draPluginClient{}
+func (v *versionedClient) v1alpha3NodePrepareResources(ctx context.Context, conn *grpc.ClientConn, req *drapbv1alpha3.NodePrepareResourcesRequest) (*drapbv1alpha3.NodePrepareResourcesResponse, error) {
+	nodeClient := drapbv1alpha3.NewNodeClient(conn)
+	return nodeClient.NodePrepareResources(ctx, req)
+}
 
-type nodeClientCreator func(addr draAddr) (
-	nodeClient drapb.NodeClient,
-	nodeClientOld drapbv1alpha2.NodeClient,
-	closer io.Closer,
-	err error,
-)
+func (v *versionedClient) v1alpha2NodeUnrepareResource(ctx context.Context, conn *grpc.ClientConn, req *drapbv1alpha2.NodeUnprepareResourceRequest) (*drapbv1alpha2.NodeUnprepareResourceResponse, error) {
+	nodeClient := drapbv1alpha2.NewNodeClient(conn)
+	return nodeClient.NodeUnprepareResource(ctx, req)
+}
+
+func (v *versionedClient) v1alpha3NodeUnrepareResources(ctx context.Context, conn *grpc.ClientConn, req *drapbv1alpha3.NodeUnprepareResourcesRequest) (*drapbv1alpha3.NodeUnprepareResourcesResponse, error) {
+	nodeClient := drapbv1alpha3.NewNodeClient(conn)
+	return nodeClient.NodeUnprepareResources(ctx, req)
+}
```

Comment:
Seeing this all laid out now, with each call only being two lines each, it seems like a bit of overkill to introduce a new type for this.

Let's drop the new type and instead just embed a single field in the existing `draPluginClient` struct called `pluginAPIVersion` to accomplish basically the same thing.

Reply:
done in 832264ac0d4